src/os/bluestore/BlueStore.cc

```diff
@@ -6325,6 +6325,40 @@ int BlueStore::statfs(struct store_statfs_t *buf)
   return 0;
 }
 
+int BlueStore::pool_statfs(uint64_t pool_id, struct store_statfs_t *buf)
+{
+  dout(20) << __func__ << " pool " << pool_id<< dendl;
+  if (!per_pool_stat_collection) {
+    dout(20) << __func__ << " not supported in a legacy mode " << dendl;
+    return -ENOTSUP;
+  }
+  buf->reset();
+  buf->total = bdev->get_size();
+  buf->available = alloc->get_free();
+
+  if (bluefs) {
+    // part of our shared device is "free" according to BlueFS
+    // Don't include bluestore_bluefs_min because that space can't
+    // be used for any other purpose.
+    buf->available += bluefs->get_free(bluefs_shared_bdev) - cct->_conf->bluestore_bluefs_min;
+
+    // include dedicated db, too, if that isn't the shared device.
+    if (bluefs_shared_bdev != BlueFS::BDEV_DB) {
+      buf->total += bluefs->get_total(BlueFS::BDEV_DB);
+    }
+  }
```

Comment:
I think the bluefs usage shouldn't be included in any pool, but only in the overall pool?